### DISCUSSION TITLE
Assigning Failover IP to Cockpit VM

### DISCUSSION THREAD

**Discussion by tgwaste** [cockpit-project/cockpit-machines#654] [General]
Hello,

I have cockpit running on OVH Cloud and have several Failover IPs. Does anyone know how I can make one of the cockpit VMs use one of the failover IPs as its IP address? I've tried a few things but cant seem to make it work.

Thank you in advance for any help.

**Comment by IndyHendrickx** [comment]
@tgwaste did u ever solve this? I have the same question.

**Comment by jelly** [comment]
I don't really understand the question. I assume you rent for example two OVH cloud servers and use something alike floating ip's where you automatically fall over to a backup VPS if you do maintenance on the other VPS. If that's the case I'm afraid it doesn't have to do much with cockpit-machines a libvirt frontend.

**Comment by KKoukiou** [comment]
@tgwaste you can assign static IP addresses to MAC addresses so that some of your VMs get always the same IP. This is possible form the network page. See DHCP entries -> add entry

https://user-images.githubusercontent.com/14921356/134897305-6f05f3ff-eaf6-4d2f-9349-6f59a1831fc3.png

**Comment by IndyHendrickx** [comment]
I managed to make it work. So ill share my findings here. To use a IP from OVH CLOUD in a Windows VM using Cockpit:

1. Login to cockpit and go to your virtual machine.
2. Add a new network interface using Direct attachment and make sure you have selected the "Generate automatically" option for MAC address.
3. Copy the generated MAC address from the newly created interface and surf to https://www.ovh.com/manager/#/dedicated/ip
4. Press the 3 dots on the right of the IP address you would like to assign and click on "Add a virtual mac"
5. Now paste the MAC address you copied in step 3, save
6. Boot into the VM and setup a static IP (.122 is used as an example)
        1. ipv4: x.y.z.122
        2. Subnet: 24
        3. gateway: x.y.z.254
        4. DNS: 8.8.8.8

Now you should be able to access the VM using VNC or RDP using the new IP address from OVH CLOUD.
